pjsip/src/pjsip-ua/sip_inv.c

```diff
@@ -3283,6 +3283,32 @@ PJ_DEF(pj_status_t) pjsip_inv_send_msg( pjsip_inv_session *inv,
 	    goto on_error;
 	}
 
+	/* Check if this is delayed manual ACK (see #416) */
+	if (tdata->msg->line.req.method.id == PJSIP_ACK_METHOD &&
+	    mod_inv.cb.on_send_ack && tdata == inv->last_ack)
```

Comment:
Will the condition **tdata == inv->last_ack** work if the ack request is created without using **pjsip_inv_create_ack** ?

Reply:
Good point.

No, it won't. ACK is supposed to be created using `pjsip_inv_create_ack()`, where `inv->last_ack` will be set and used by ACK auto retransmission, unfortunately the `on_send_ack` docs does not mention it either (to be updated).

Btw, is there any sample case that app does not want to (or even cannot) use `pjsip_inv_create_ack()`?

Reply:
I cannot think of any case where you wouldn't use the provided API to construct such a message. The only thing is that if it **has** to be used, then, IMHO, it should be mentioned somewhere.

But as you said in your comment, if the doc is updated with this piece of information, then it's OK !